ballerina/listener_oauth2_provider.bal

```diff
@@ -278,3 +278,20 @@ isolated function validateFromCache(cache:Cache oauth2Cache, string token) retur
     }
     return;
 }
+
+isolated function parseExpClaim(json expClaim) returns int|() {
+    if expClaim is int {
+        return expClaim;
+    } else if expClaim is string {
+        int|error parsedInt = int:fromString(expClaim);
+        if parsedInt is int {
+            return parsedInt;
+        } else {
+            log:printError("Failed to parse string to integer for exp field", parsedInt);
+            return ();
+        }
+    } else {
+        log:printError("Invalid type for exp field, expected int or string");
+        return ();
+    }
+}
```

Comment:
```suggestion
}

```

Reply:
Was this a mistake or is this change necessary? @MohamedSabthar

Reply:
Need to end each file with a new line @ovindu-a

Reply:
Ohh right